lib/index.ts

```diff
@@ -23,16 +26,55 @@ function inspect(root: string, targetFile?: string, options?: any) {
         dockerImageId: result[1].imageId,
       };
       const pkg: any = result[1].package;
+      const dockerfilePackages = result[2]
+        ? getDockerfileDependencies(result[2].dockerfilePackages,
+                                    pkg.dependencies)
+        : [];
+
       pkg.docker = pkg.docker || {};
-      pkg.docker.baseImage = result[2];
       pkg.docker.binaries = result[1].binaries;
+      pkg.docker = {
+        ...pkg.docker,
+        ...result[2],
+        dockerfilePackages,
+      };
+
       return {
         plugin: metadata,
         package: pkg,
       };
     });
 }
 
+function getDockerfileDependencies(dockerfilePackages: DockerFilePackages,
+                                   dependencies) {
+  for (const dependencyName in dependencies) {
+    if (dependencies.hasOwnProperty(dependencyName)) {
+      const sourceOrName = dependencyName.split('/')[0];
+      const dockerfilePackage = dockerfilePackages[sourceOrName];
+
+      if (dockerfilePackage) {
+        collectDeps(dependencies[dependencyName]).forEach((dep) => {
```

Comment:
Let's say package `a` is installed by the base image, and package `b` is installed by the docker file.
As I understand the code, it means that if a package `b` is depending on `a` it will mark `a` as a docker file package - which will mark a vuln on this package as a docker file vuln. LMK what do you think.

Reply:
Yup, it will do exactly this. Any suggestions on improving it?